### PR TITLE
[lua] Use native bit operators for Int32 wrapping on Lua 5.3+

### LINKED ISSUE #11013: [lua] Int32/64 overflows incorrect
Int64 throws error:

изображение

Int32 does not overflow:

изображение

### PR DESCRIPTION
**Pull request by jdonaldson**
## Summary
Fix `haxe.Int32` overflow wrapping on Lua 5.3+ when no bit library is explicitly loaded.

## Problem
When using `haxe.Int32` on Lua 5.3+ without the `bit32` library installed, overflow didn't wrap properly:

```haxe
var i32:haxe.Int32 = 2147483647;
var wrapped = i32 + 1;
trace(wrapped);  // Was: 2147483647 (clamped), Should be: -2147483648 (wrapped)
```

This happened because:
1. `_hx_bit.lua` (which loads bit32/bit libraries) is only included when explicit bit operations are used
2. `_hx_bit_clamp.lua` checks for `_hx_bit_raw` and falls back to clamping if unavailable
3. On Lua 5.3+, native bit operators (`&`, `|`, etc.) are available but weren't being used

## Fix
Updated `_hx_bit_clamp.lua` to detect and use native Lua 5.3+ bit operators when `_hx_bit_raw` is not available:
- **Lua 5.3+**: Uses native `&` operators for proper 32-bit wrapping
- **Lua 5.1/5.2 with bit32/bit**: Uses library (unchanged)
- **Lua 5.1/5.2 without library**: Falls back to clamp (unchanged, same as before)

## Note on Int64
The original issue also mentions `cast 1` to Int64 failing. This is expected Haxe behavior - `cast` is a type assertion, not a conversion. Use `haxe.Int64.ofInt(1)` instead.

Partially fixes #11013

### CODE REVIEW

**Comment by Simn** [comment]
This also needs a test.

**Comment by tobil4sk** [comment]
Can we prefer the native operators over bit32 if they're available?

Currently we install bit32 on lua 5.4 because it doesn't have the builtin library anymore, but if we use it by default then the "test" can just be not installing bit32 on lua 5.4.

EDIT: Actually, if there is incorrect clamping behaviour, that should be tested explicitly too.

**Comment by tobil4sk** [comment]
Linking some relevant issues: #8852, #9412, #9475.

**Comment by tobil4sk** [comment]
Actually, Int32 overflow behaviour is specified across targets so it should actually be covered in https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/haxe/Int32.unit.hx.

What we need here is a test for the the overflow in the specific case where bit32 is not loaded. Since the unit tests are always run with bit32 available as a builtin or external library, and _hx_bit.lua is always included, adding more unit tests doesn't really verify anything.

This could be a standalone misc test (`tests/misc/lua/projects`) so that we have more control over what is included. We could run it like this to disable `require` completely, making sure it works without loading bit32:
`lua -e "require = nil" main.lua`

File: std/lua/_lua/_hx_bit_clamp.lua [hunk 1]
```diff
@@ -10,6 +30,7 @@ if _hx_bit_raw then
     return _hx_bit_raw.band(v, 2147483647 ) - math.abs(_hx_bit_raw.band(v, 2147483648))
     end
 else
+    -- Fallback for Lua 5.1/5.2 without bit library: clamp instead of wrap
```
Comment (tobil4sk) [inline std/lua/_lua/_hx_bit_clamp.lua]: On lua 5.2, bit32 is builtin so it should always be available. However, `_hx_bit_raw` is only set if _hx_bit.lua is included, so we may end up using the fallback implementation which is undesirable. Similarly, on luajit, bit is builtin.

_hx_bit and _hx_bit_clamp used to be included together originally, so `_hx_bit_raw` was always accurate, but it changed in: d32f65da888704d0caa2f161de231ae5dd041672 (also relevant: 229476be6434a81cfec852137bef1e645ffcd3d2)

With #12587, I made it so that _hx_bit won't error immediately if no bit library is found, so it will be safe to include unconditionally so the `_hx_bit_raw` check can be accurate again accurate.

**Comment by tobil4sk** [comment]
Maybe we could also fix the fallback implementation by doing some trickery with `%`?
